modal/functions.py

```diff
@@ -1388,8 +1392,22 @@ async def get(self, timeout: Optional[float] = None) -> R:
 
         The returned coroutine is not cancellation-safe.
         """
+
+        if self._is_generator:
+            raise Exception("Cannot get the result of a generator function call. Use `iter_gen` instead.")
+
         return await self._invocation().poll_function(timeout=timeout)
 
+    async def iter_gen(self) -> AsyncGenerator[Any, None]:
```

Comment:
Maybe we should call it `get_gen()` instead? As a corollary to `Function.remote()` and `Function.remote_gen()`

Reply:
Wdyt @aksh-at ? I have a vague memory we discussed adding generator support to FunctionCall like ~2 years ago 😂

Reply:
Yep makes sense!